### DISCUSSION TITLE
Any plans to support SSG like Gatsby?

### DISCUSSION THREAD

**Discussion by kawpii** [solidjs/solid#595] [General]
I know that Solid [kind of support SSG ](https://github.com/solidjs/solid/blob/ce013b658470de45b3980fcf6c4c6bee12c4c9c7/packages/solid-ssr/README.md) but not much optimizations like Gatsby.

- Image optimization
- Fast page navigation with [Link](https://www.gatsbyjs.com/docs/reference/built-in-components/gatsby-link/)
- Resource pre-fetching
- Per route code splitting
- Inlining Critical CSS
- Plugins

For JAM stack type websites using Solidjs we have to do lot of manual work unlike Gatsby. Since demand for JAM stack on the rise, Gatsby like SSG tool for Solidjs will be highly beneficial.

@ryansolid  What do you think?

**Comment by ryansolid** [comment]
Yeah many of these pieces are possible using some combination of libraries and manual configuration. That all being said while I'd welcome this sort of exploration, I imagine most of this comes from the community over time. Think of Solid like React. There is a lot of work to be done on improving the core and enabling developers to do more with Solid with things they couldn't do on their own. I am also working when I have time on a starter(SolidStart) using Vite that will support SSG but it is unclear when a lot of the specific stuff gets attention.

Things that I do seeing addressed in the shorter-term:
* Fast navigation/Link preloading of JS assets could be built into the router. Data could be too but I think it might be too hard to generalize and probably belongs in specific starters. It's likely the SolidStart starter will have this.
* Per route code splitting is pretty trivial to do with the `lazy` component today. This would be the responsibility of the starter to handle the file-based routing to do so automatically. The one I'm building for SolidStart already does this.

The rest will probably take more time. Using Vite, Plugins are basically already available.

But beyond this realistically I wouldn't be expecting too much in this direction outside of a community-driven effort. There is enough to support with a small core team between the core and website/playground etc.

**Reply by kawpii** [reply]
Thanks for the response!

> But beyond this realistically I wouldn't be expecting too much in this direction outside of a community-driven effort. There is enough to support with a small core team between the core and website/playground etc.

Understood, community is the way.
